connectors/sources/sharepoint_online.py

```diff
@@ -1750,38 +1756,50 @@ async def get_docs_incrementally(self, sync_cursor, filtering=None):
                         yield list_item, download_func, OP_INDEX
 
                 # Sync site pages
-                async for site_page in self.site_pages(site, site_access_control):
+                async for site_page in self.site_pages(
+                    site, site_access_control, check_timestamp=True
+                ):
                     # Always include site admins in site page access controls
                     site_page = self._decorate_with_access_control(
                         site_page, site_admin_access_control
                     )
                     yield site_page, None, OP_INDEX
 
+        self.update_sync_timestamp_cursor(timestamp)
+
     async def site_collections(self):
         async for site_collection in self.client.site_collections():
             site_collection["_id"] = site_collection["webUrl"]
             site_collection["object_type"] = "site_collection"
 
             yield site_collection
 
-    async def sites(self, hostname, collections):
+    async def sites(self, hostname, collections, check_timestamp=False):
         async for site in self.client.sites(
             hostname,
             collections,
             enumerate_all_sites=self.configuration["enumerate_all_sites"],
             fetch_subsites=self.configuration["fetch_subsites"],
         ):  # TODO: simplify and eliminate root call
-            site["_id"] = site["id"]
-            site["object_type"] = "site"
+            if not check_timestamp or (
+                check_timestamp
+                and site["lastModifiedDateTime"] >= self.last_sync_time()
+            ):
+                site["_id"] = site["id"]
+                site["object_type"] = "site"
 
-            yield site
+                yield site
```

Comment:
Just to double check - if an object is uploaded to Site's Drive, the root Site's modified date is not changed.

Because of it, the approach here will ignore resyncing the Site Drive?

Reply:
> if an object is uploaded to Site's Drive, the root Site's modified date is not changed.

It actually _does_ change the root Site's modified date. I tested this with drive items, lists, list items, and site page edits. So for everything we care about, the change date propagates up. List item changes update the List's modified date, too.

The only case where nesting updates don't seem to apply is subsite changes don't modify site dates. However, since we're enumerating sites and subsites together in the same generator, this isn't problematic - I can exclude a top-level site but still enumerate the subsite.

Reply:
What about permission changes? Changes in dynamic groups?

I have a weird feeling, that it's possible that there will be weird edge-cases that will be problematic :/

Reply:
Sharing changes also resulted in changing the `lastModifiedDateTime`. Changes to dynamic groups won't. But today in a full sync, those wouldn't get picked up either, since we'd detect that the `_id` and `_ts` have not changed. It's something we need to address in : https://github.com/elastic/connectors-python/issues/1664